src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestStream.cs

```diff
@@ -165,7 +165,26 @@ private void ValidateState()
                 case FrameStreamState.Closed:
                     throw new ObjectDisposedException(nameof(FrameRequestStream));
                 case FrameStreamState.Aborted:
-                    throw new IOException("The request has been aborted.");
+                    throw new OperationCanceledException("The request has been aborted.");
```

Comment:
Is it wrong to throw a `TaskCanceledException` from a synchronous call to Read or Write? It is an `OperationCanceledException` so the handling code should be the same. I think it's best to have one ValidateState method if possible.